### ISSUE TITLE
NDLAr track unit vectors are flipped

### ISSUE THREAD

**Issue by clarenc3** [DUNE/ND_CAFMaker#14]
The NDLAr `enddir` entries are always negative in the `z` direction, which seems like a bug that could just be fixed by a `-1*` operation. But this should be understood, assigning Jeremy in the meantime.

The direction default also appears to be zero, which feels like a very dangerous default value (seeing that it's a completely valid direction for a unit vector to have). 

Whilst I'm on it, the NDLAr `dir` is not filled.

**Comment by clarenc3** [comment]
Attaching comparison of z unit vector for NDLAr and TMS to better illustrate the issue
![Screenshot from 2022-08-19 15-27-52](https://user-images.githubusercontent.com/11278399/185684317-e22776fb-8073-4524-bf5e-6c32c02adb23.png)

![Screenshot from 2022-08-19 15-28-17](https://user-images.githubusercontent.com/11278399/185684328-05583549-b5af-4cb6-be0a-38cf9939a83f.png)

**Comment by chenel** [comment]
I did some digging, and it looks like [I already fixed this issue](https://github.com/chenel/dune-nd-lar-reco/commit/8335a9e74a98c49de6f6b8c0e0b1fae33d08ad78), but it didn't make it into the [version of the ND-LAr-reco driver used for the workshop CAFs](https://github.com/chenel/dune-nd-lar-reco/releases/tag/2022_lbl_wksp_cafs).

If there are any respins pending, we should certainly update.  Otherwise we'll pick it up the next time...

**Comment by chenel** [comment]
> The direction default also appears to be zero, which feels like a very dangerous default value (seeing that it's a completely valid direction for a unit vector to have).

We could certainly change it to be `(NaN, NaN, NaN)`, which I'm sure @cmmarshall  will love 😄

**Comment by chenel** [comment]
> Whilst I'm on it, the NDLAr dir is not filled.

Yeah, that's right.  I think we added `dir` for you to use, actually, not because I was trying to write it out 😄   If we think it's useful to have that as well I could put it on my list to store it, but it requires adding some stuff.

**Comment by clarenc3** [comment]
I also found that many tracks that clearly have a start and end point in the NDLAr don't have the enddir field filled either; is that connected to this? See attached screenshot, where I to TTree:Scan showing the saved unit vector of the direction vector (enddir) and the start and end point of the track. 
![Screenshot from 2022-08-19 20-51-04](https://user-images.githubusercontent.com/11278399/186213941-c5acdbb6-cb5c-4cf7-a7e5-7023b08bdb03.png)

**Comment by chenel** [comment]
I think there's a chance this is self-resolving with my updated suite of commits, but it'll take testing to prove that.  Maybe @andrewmogan and I can make an updated file or two and see

**Comment by chenel** [comment]
Ok.  Andrew and I have rerun the reco on one of the files from the LBL workshop, and I verified that the negative `enddir` entries are resolved with the update `dune-nd-lar-reco`.  The missing `enddir` rows also do not appear to happen.  So I consider those resolved.


I have also opened #15, which will fill `SRTrack::dir`.  This won't do anything useful until we rerun the ND-LAr reco summary files for input, but it's now there, so once that's merged, I think we can close this issue.
